eisbuk-admin/cypress/support/commands.js

```diff
@@ -36,28 +37,41 @@ connectFunctionsEmulator(functions, "localhost", 5001);
 
 /**
  * Set up app for testing and log in as default admin
+ * @param {boolean} doLogin - whether to log in as default admin
  */
-Cypress.Commands.add("initAdminApp", () => {
+Cypress.Commands.add("initAdminApp", async (doLogin = true) => {
   // create a random organization name in order to run each test
   // against it's own organization, using the same db without conflicts
   const organization = uuidv4();
+  console.log(doLogin);
+  debugger;
   cy.on("window:before:load", (win) => {
     // the `organization` name is set to local storage and read from the app while testing
     win.localStorage.setItem("organization", organization);
   });
+  // create a default organization (containing the default user as an admin)
+  await httpsCallable(
+    functions,
+    CloudFunction.CreateOrganization
+  )({ organization });
 
-  // create or login a default user in order to run the tests with admin access
-  createUserWithEmailAndPassword(
+  // Always create a user
+  await createUserWithEmailAndPassword(
     auth,
     defaultUser.email,
     defaultUser.password
-  ).catch(() => {
-    signInWithEmailAndPassword(auth, defaultUser.email, defaultUser.password);
+  ).catch(async () => {
+    if (doLogin) {
+      await signInWithEmailAndPassword(
+        auth,
+        defaultUser.email,
+        defaultUser.password
+      );
+      console.log(`Logged in as ${defaultUser.email}`);
+    } else {
+      debugger;
+      await signOut(auth);
+      console.log(`Logged out`);
+    }
```

Comment:
I don't quite understand how this works...From my understanding, the login should be ran only if the `createUserWithEmailAndPassword` throws (the user already exists). Following that logic, if createUser(...) is successful, the `if (doLogin)` won't be ran and (from my understanding) the created user will remain logged in (as no `signOut` is ran)...but then again the (UI) login test would not be possible after this, but still it passes...

Could you please explain this

Reply:
Right! It was working because the non-catch path only happens on the first test, since subsequent tests will find an already existing user.
I fixed the function converting the `.catch()` to use `try {} catch()` instead. The problem you mentioned should be fixed, and it's now more readable. There was also a spurious `debugger` statement.